Automattic/wp-calypso
# Launch sites: Add a launch site button

#### Changes proposed in this Pull Request

This adds the ability to launch a site from within Calypso. It relies on this server side patch: code-D19843, which adds a new endpoint.

The visibility settings are replaced with a "launch site" button on site settings, until the site is launched:
<img width="907" alt="screenshot 2018-10-26 at 13 35 51" src="https://user-images.githubusercontent.com/275961/47566577-18a02a80-d924-11e8-9f27-51a91a41c944.png">

When the user launches the site, the site settings update and display the normal visibility settings options:
<img width="829" alt="screenshot 2018-10-26 at 13 36 57" src="https://user-images.githubusercontent.com/275961/47566635-3b324380-d924-11e8-84f9-6ec55e539034.png">


#### Testing instructions
1. Apply code-D19843 to your sandbox
1. Create a new site from /start/private
1. Confirm that the site is unlaunched by checking for the banner on the frontend of the site
1. Open site settings for that site (http://calypso.localhost:3000/settings/general/<site>)
1. Confirm that you see the launch site button (as above)
1. Click the launch site button (You should see a notice telling you the site is launching)
1. In a moment the page will update and you'll see the visibility settings of the site and a success message (as above)

### Notes
I considered adding a reducer for this, and also another action (`SITE_LAUNCH_RECEIEVE`) to manage the disabled state of the "Launch site" button, but it felt like adding a lot of extra code for very little benefit. If we think it is worthwhile then I'm happy to add it (maybe in a new PR) but I wanted to keep the scope of this tight.
